writer.go

```diff
@@ -0,0 +1,145 @@
+package pinecone
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/pinecone-io/go-pinecone/pinecone"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// Writer implements a writer logic for Sap hana destination.
+type Writer struct {
+	client *pinecone.Client
+	index  *pinecone.IndexConnection
+}
+
+// NewWriter New creates new instance of the Writer.
+func NewWriter(ctx context.Context, config DestinationConfig) (*Writer, error) {
+	sdk.Logger(ctx).Trace().Msg("Creating new writer.")
+
+	sdk.Logger(ctx).Error().Msgf("API: %v, INDEX:%v", config.PineconeAPIKey, config.PineconeHostURL)
+
+	pineconeClient, indexConnection, err := NewPineconeClient(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create pinecone pineconeClient: %v", err)
```

Comment:
We should use `%w` here to wrap the error (same in other places too).

Reply:
Also refactored that in another pr.